app/routines/openstax/accounts/create_temp_account.rb

```diff
@@ -0,0 +1,36 @@
+# Routine for creating a temporary account by email or username
+
+module OpenStax
+  module Accounts
+    class CreateTempAccount
+
+      lev_routine
+            
+      protected
+
+      def exec(inputs={})
+        result = Api.create_temp_account(inputs)
+        fatal_error(code: :invalid_inputs) unless result.success?
+
+        struct = OpenStruct.new
+        Api::V1::UnclaimedAccountRepresenter.new(struct).from_json(result)
+        id = struct.id
+
+        account = Account.find_or_initialize_by(openstax_uid: id)
+
+        unless account.persisted?
+          username = inputs[:username]
+          while username.nil? || Account.where(username: username).exists? do 
+            username = SecureRandom.hex(3).to_s
```

Comment:
I don't think a random username is all the usable, so can we just not have one and maybe return "unknown_user" or something if some other part of the code asks?  @kevinburleigh75 ?